x-pack/libbeat/reader/parquet/parquet_test.go

```diff
@@ -0,0 +1,303 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//nolint:errcheck // It's a test file, we don't care about errors here.
+package parquet
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func BenchmarkReadParquetSingleSerialBatch_1000(b *testing.B) {
+	f := "testdata/taxi_2023_1.parquet"
+
+	cfg := &Config{
+		ProcessParallel: true,
+		BatchSize:       1000,
+	}
+
+	path, _ := filepath.Abs(f)
+	file, err := os.Open(path)
+	if err != nil {
+		b.Fatalf("failed to open parquet file: %v\n", err)
+	}
+	defer file.Close()
+
+	fn, size := getFileData(b, file)
+	var batches int
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		file.Seek(0, 0)
+		batches = readParquet(b, cfg, file)
```

Comment:
You can apply table driving testing to benchmarks. I recommend that here. I think you could get this down to one reusable benchmark function that accepts the parameters from your test matrix. It looks like the three axes in the matrix should be

- input parquet file (string)
- ProcessParallel (bool)
- BatchSize (int)

Next, think about what the basic operation ("op") should be in your benchmarks. Is the op reading an entire file or is the op reading a single row? I think it would be meaningful to measure the op as reading a single row. It might also be interesting to benchmark the construction cost of `NewStreamReader` by itself too to understand the memory costs of construction.